Added ability to refresh tokens and re-authenticate using refresh token

tycrek opened the pull request:
Essentially a re-do of #15. This allows users to refresh tokens; save a refresh token for later use; and authenticate using a refresh token.

To refresh the tokens, call:

```java
// Initialize MsaAuthenticationService and authenticate somehow
auth.login();

// Some time later when the access token has expired
MsaAuthenticationService.MsTokenResponse response = auth.refreshToken();
System.out.println(response.refresh_token);
```

In order to authenticate using a previously saved refresh token, you must first set only the **username** and **refresh token** for a `MsaAuthenticationService` like so:

```java
MsaAuthenticationService auth = new MsaAuthenticationService(AZURE_CLIENT_ID);
auth.setUsername(USERNAME);          // A Microsoft account email
auth.setRefreshToken(REFRESH_TOKEN); // Refresh token from previous step
auth.login(); // Internally authenticates using the refresh token
```

Commit:
Added `offline_access` scope for re-authenticating

Commit:
Don't use hardcoded client ID here

Commit:
Added ability to refresh tokens and re-authenticate using refresh tokens

Camotoy commented on src/main/java/com/github/steveice10/mc/auth/service/MsaAuthenticationService.java:
```diff
@@ -295,7 +343,7 @@ public String toString() {
         protected MsCodeRequest(String clientId) {
             this.client_id = clientId;
-            this.scope = "XboxLive.signin";
+            this.scope = "XboxLive.signin offline_access";
```
Does this break existing functionality in any way?

tycrek replied:
Nope, doesn't break anything. Only really noticable change is what appears in the permissions page: ![new permission with `offline_access`](https://ass.rip/Y7Pygn6y5uhu/direct)

Camotoy replied:
Are you able to make this a boolean for those who don't need offline access?

tycrek replied:
Added overloaded constructor with a boolean parameter to set this flag. Should help avoid breaking existing users projects too as this value will be set to `false` in the existing constructor

Commit:
MsCodeRequests now accepts a boolean to set `offline_access`
This boolean is set to false by default via overloaded constructor to
avoid breaking existing users projects.
